packages/filebrowser/style/base.css

```diff
@@ -215,13 +215,36 @@
 .jp-DirListing-item {
   display: flex;
   flex-direction: row;
-  padding: 4px 12px;
+  align-items: center;
+  padding-right: 12px;
   -webkit-user-select: none;
   -moz-user-select: none;
   -ms-user-select: none;
   user-select: none;
 }
 
+.jp-DirListing-checkboxWrapper {
+  /* Increases hit area of checkbox. */
+  padding: 4px;
+}
+
+.jp-DirListing-checkboxWrapper.jp-mod-visible {
+  visibility: visible;
+}
+
+/* For devices that support hovering, hide checkboxes until hovered, selected...
+*/
+@media (hover: hover) {
```

Comment:
I tested that this works on the following:

- iPad -> checkboxes always visible
- MacBook -> checkboxes visible on hover
- Windows laptop -> checkboxes visible on hover
- Windows laptop in tablet mode -> checkboxes always visible

Reply:
Thanks for the extensive testing!